src/uiHelpers_nanox.c

```diff
@@ -0,0 +1,110 @@
+#include <os_io_seproxyhal.h>
+#if defined(TARGET_NANOX)
+#include "uiHelpers.h"
+
+// Helper macro for better astyle formatting of UX_FLOW definitions
+#define LINES(...) { __VA_ARGS__ }
+
+// Flow scrolling
+void scrolling_confirm()
+{
+	TRY_CATCH_UI({
+		assert_uiScrolling_magic();
+		scrollingState_t* ctx = scrollingState;
+		uiCallback_confirm(&ctx->callback);
+	});
+}
+
+UX_FLOW_DEF_VALID(
+        ux_display_scrolling_flow_1_step,
+        bnnn_paging,
+        scrolling_confirm(),
+        LINES(
+                (char *) &displayState.scrolling.header,
+                (char *) &displayState.scrolling.fullText
+        )
+);
+
+// Flow confirm
+void confirm_confirm()
+{
+	TRY_CATCH_UI({
+		assert_uiConfirm_magic();
+		confirmState_t* ctx = confirmState;
+		uiCallback_confirm(&ctx->callback);
+	});
+}
+
+UX_FLOW_DEF_VALID(
+        ux_display_confirm_flow_1_step,
+        pbb,
+        confirm_confirm(),
+        LINES(
+                &C_icon_validate_14,
+                (char *) &displayState.scrolling.header,
+                (char *) &displayState.scrolling.currentText,
+        )
+);
+
+// Flow refect
```

Comment:
Typo. Plus this ins't reject flow. It's part of "confirm" flow. We should really rename confirm->prompt and scrolling->text. And have displayPrompt and displayText calls

Reply:
done